pkg/api/v1alpha1/nutanixdatacenterconfig.go

```diff
@@ -44,3 +50,13 @@ func GetNutanixDatacenterConfig(fileName string) (*NutanixDatacenterConfig, erro
 	}
 	return &clusterConfig, nil
 }
+
+// SetDefaults sets default values for the NutanixDatacenterConfig object.
+func (c *NutanixDatacenterConfig) SetDefaults() {
+	if c.Spec.CredentialRef == nil {
+		c.Spec.CredentialRef = &Ref{
+			Kind: constants.SecretKind,
+			Name: constants.NutanixCredentialsName,
+		}
+	}
+}
```

Comment:
Can we move this to `nutanixdatacenterconfig_types.go`? Looks like all the other `SetDefaults()` are in `_types.go`